aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1168,16 +1179,54 @@ impl AptosVM {
                     &mut new_published_modules_loaded,
                     &storage_gas_params.change_set_configs,
                 ),
-            TransactionPayload::Multisig(payload) => self.execute_multisig_transaction(
-                resolver,
-                session,
-                gas_meter,
-                &txn_data,
-                payload,
-                log_context,
-                &mut new_published_modules_loaded,
-                &storage_gas_params.change_set_configs,
-            ),
+            TransactionPayload::Multisig(payload) => {
+                if !self.is_simulation {
+                    self.execute_multisig_transaction(
+                        resolver,
+                        session,
+                        gas_meter,
+                        &txn_data,
+                        payload,
+                        log_context,
+                        &mut new_published_modules_loaded,
+                        &storage_gas_params.change_set_configs,
+                    )
+                // Otherwise this is simulation.
+                // TODO: de-duplicate with 'execute_multisig_transaction'.
```

Comment:
If that is a duplicate, I would pull it for symmetry in outs own function.

Reply:
Yep, this is the code I have not refactored due to different flow, having it in its own section is a good point